src/main/java/io/jenkins/plugins/checks/github/GitHubChecksPublisher.java

```diff
@@ -127,4 +147,68 @@ private GHCheckRunBuilder applyDetails(final GHCheckRunBuilder builder, final Gi
         return builder;
     }
 
+    private boolean orgCheck(String username, GitHub github, boolean isPrivate) throws IOException {
+        // skip org check if the repo is private
+        if (isPrivate) {
+            return true;
+        }
+        try {
+            GHUser user = github.getUser(username);
+            GHOrganization org = github.getOrganization(GITHUB_ORG);
+            boolean isConfluentinc = user.isMemberOf(org);
+            buildLogger.log("contributorisConfluentinc: " + isConfluentinc);
+            return isConfluentinc;
+        }
+        catch (IOException e) {
+            buildLogger.log("Failed to connect to GitHub " + e);
+            return false;
+        }
+    }
+
+    /**
+     * Whether to publish github status.
+     *
+     * @param isPrivate
+     *         whether repo is private
+     * @param inGithubOrg
+     *         whether contributor is member of confluentinc
+     * @param publishNonConfluentIncPR
+     *         publish non-confluentinc PR status only
+     * @param publishConfluentIncPR
+     *         publish confluentinc PR status only
+     * @return true if we should publish github status
+     */
+    private boolean shouldPublish(boolean isPrivate,
+                         boolean inGithubOrg,
+                         boolean publishNonConfluentIncPR,
+                         boolean publishConfluentIncPR) {
+        if (isPrivate) {
+            // if repo is private, do not skip
+            return true;
+        }
+        else if (!publishConfluentIncPR && !publishNonConfluentIncPR) {
+            return false;
+        }
+        else {
+            // if contributor is member of confluentinc
+            if (inGithubOrg) {
+                if (publishConfluentIncPR) {
+                    return true;
+                }
+                else if (publishNonConfluentIncPR) {
+                    return false;
+                }
+            }
+            else {
+                if (publishNonConfluentIncPR) {
+                    return true;
+                }
+                else if (publishConfluentIncPR) {
+                    return false;
+                }
+            }
+        }
+        return true;
```

Comment:
should the fall-through case here be `false`?

Reply:
I was thinking the default behavior should be always publish?

Reply:
is this entire `else` block & fall-through equivalent to:

```
return (inGithubOrg && publishConfluentIncPR) || (!inGithubOrg && publishNonConfluentIncPR)
```
because from the previous `else if`, we know that one of `publishConfluentIncPR` or `publishNonConfluentIncPR` is `true`